kosogor-terraform/src/main/kotlin/tanvd/kosogor/terraform/tasks/validate/ValidateModulesTask.kt

```diff
@@ -45,36 +46,41 @@ open class ValidateModulesTask : DefaultTask() {
                         init(workingDir)
                         validate(workingDir)
                     } finally {
-                        File(file.parentFile, "terraform").deleteRecursively()
+                        File(file.parentFile, ".terraform").deleteRecursively()
                         providerFile?.delete()
                     }
                     println("Validated module ${packageInfo.group}:${packageInfo.name}:${packageInfo.version}")
                 }
     }
 
     private fun init(workingDir: File) {
-        val curInitDir = File(workingDir, "terraform")
+        val curInitDir = File(workingDir, ".terraform")
+        if (curInitDir.exists()) curInitDir.deleteRecursively()
         if (terraformDsl.validater.cacheInitPlugins && GlobalFile.tfInitDir.exists()) {
-            FileUtils.copyDirectory(GlobalFile.tfInitDir, curInitDir)
+            curInitDir.mkdirs()
+            File(GlobalFile.tfInitDir, "plugins").walk().forEach {
+                val targetFile = File(curInitDir, it.toRelativeString(GlobalFile.tfInitDir))
+                Files.copy(it.toPath(), targetFile.toPath(), COPY_ATTRIBUTES)
+            }
         }
+        CommandLine.executeOrFail(GlobalFile.tfBin.absolutePath, listOf("init"), workingDir, false)
 
         if (terraformDsl.validater.cacheInitPlugins && !GlobalFile.tfInitDir.exists()) {
-            val retInit = CommandLine.execute(GlobalFile.tfBin.absolutePath, listOf("init"), workingDir, false)
-            if (retInit != 0) {
-                error("Terraform init failed")
-            }
-
             GlobalFile.tfInitDir.mkdirs()
-            FileUtils.copyDirectory(File(curInitDir, "plugins"), File(GlobalFile.tfInitDir, "plugins"))
+            File(curInitDir, "plugins").walk().forEach {
```

Comment:
Probably we can move that copy into something like `FileUtils.copyWithAttrs`? And remove duplication :)